### PR TITLE
Add some detail to handbook about different theme types

### PR DESCRIPTION
**Pull request by glendaviesnz**
## What?
Add some detail to the handbook about the different types of themes and how they can interact wit the editor

## Why?
In recent discussion around some of the theme styling issues in 5.9 there appears to be a lack of a shared understanding of what different theme types are referred to and how they can interact with the editor

## How?
Adds brief details about Classis, Hybrid, and Block theme types.

## Testing Instructions
Read it and confirm technical correctness, grammar, spelling

### CODE REVIEW

File: docs/how-to-guides/themes/README.md [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+
+### Hybrid theme
+
+As well as using [Theme Supports](/docs/how-to-guides/themes/theme-support.md) to configure the block editor and block content, a `Classic` theme can also include a [theme.json](/docs/how-to-guides/themes/theme-json.md) file to achieve this. A theme does not have to be a `Block` theme in order to take advantage of some of the flexibility provided by the use of a `theme.json` file.
+
+### Block theme
+
+This is any theme that has, at a minimum, an `index.html` format template in the `/block-templates` or `/templates` folders, and with templates  provided in form of block content markup. While many `Block` themes will make use of a `theme.json` file to provide configuration and styling settings a `theme.json` is not a requirement of `Block` themes. The advantage of `Block` themes is that the block editor to be used to edit all areas of the site: headers, footers, sidebars, etc.
```
Comment (andrewserong) [inline docs/how-to-guides/themes/README.md]: Tiny grammar fix:

```suggestion
This is any theme that has, at a minimum, an `index.html` format template in the `/block-templates` or `/templates` folders, and with templates  provided in form of block content markup. While many `Block` themes will make use of a `theme.json` file to provide configuration and styling settings a `theme.json` is not a requirement of `Block` themes. The advantage of `Block` themes is that the block editor can be used to edit all areas of the site: headers, footers, sidebars, etc.
```

File: docs/how-to-guides/themes/README.md [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
-The block editor provides a number of options for theme designers and developers, including theme-defined color settings, font size control, and much more.
+The block editor provides a number of options for theme designers and developers, to interact with it, including theme-defined color settings, font size control, and much more.
+
+## Types of themes
+
+### Classic theme
+
+In terms of block editor terminology this is any theme that defines its templates in the traditional `.php` file format, and that doesn't have an `index.html` format template in the `/block-templates` or `/templates` folders. A `Classic` theme still has the ability to provide configuration and styling options to the block editor, and block content, via [Theme Supports](/docs/how-to-guides/themes/theme-support.md). 
+
+### Hybrid theme
```
Comment (glendaviesnz) [inline docs/how-to-guides/themes/README.md]: @annezazu, @youknowriad  - have we got this terminology correct in terms of a Classic theme that has a theme.json file? We thought `Hybrid` was a term that was being used, but weren't 100% sure of that.

Comment (glendaviesnz) [reply docs/how-to-guides/themes/README.md]: The other approach is to not name this combination, and just note under Classic theme that they can provide config via either Theme Supports or a theme.json

File: docs/how-to-guides/themes/README.md (hunk 1 above)
Comment (glendaviesnz) [reply docs/how-to-guides/themes/README.md]: Thanks, fixed

File: docs/how-to-guides/themes/README.md (hunk 2 above)
Comment (youknowriad) [reply docs/how-to-guides/themes/README.md]: The definition of a hybrid theme is different that what is mentioned here. A hybrid theme is a theme that has both "php templates" and "block templates", meaning depending on the WP version can work as a block theme or classic theme (block theme in newer versions).

I think though that it's probably not worth mentioning in the docs. I don't expect it to be so common.

**Review by andrewserong — APPROVED** [review]
Reads well to me, thanks for updating the docs!

**Pull request merged**